Experimental/MultiplayerUseCases/Assets/UseCaseSamples/NetvarVsRpc/Scripts/ColorManager.cs

```diff
@@ -65,23 +65,23 @@ void Update()
 
         void OnClientRequestColorChange()
         {
-            OnServerChangeColorServerRpc();
+            ServerChangeColorRpc();
         }
```

Comment:
@RikuTheFuffs Not sure if this has been raised before, but why not just invoke the Rpc to server here?

Reply:
@RikuTheFuffs

Reply:
In this specific case you could invoke the RPC directly since it's just one line, however it is cleaner to have a method that contains all that the client needs to do to request a color change for several reasons:

1. Makes the code easier to test automatically, as you'd only have to expose one method
2. That method could be called from multiple places
3. There could be more than one line in that method

Reply:
Could you elaborate how that would look like for point 1?
What if it is invoked from multiple places? Other places in code could also invoke the Rpc method, no?
As for the third point, I would favour brevity and simplicity in this case. It's a shell method at the moment, and if the code for any reason changes in the future, we should refactor it then.

Reply:
> Could you elaborate how that would look like for point 1?
You just make OnClientRequestColorChange `internal` instead of private and call it.

> What if it is invoked from multiple places? Other places in code could also invoke the Rpc method, no?

If the method is 2+ lines, you have to call both lines everywhere. Nothign changes when it's just 1line, as I mentioned in my previous comment.

> As for the third point, I would favour brevity and simplicity in this case. It's a shell method at the moment, and if the code for any reason changes in the future, we should refactor it then.

I disagre: this is an educational project, so shouldn't we educate on how to do things right?
And "doing things right" means encapsulating the logic so that you don't have to care about the underlying implementation.

If you don't go that way, what happens is that in a real project with multiple contributos you end up with peopel copy-pasting stuff around, which quickly becomes a nightmare for both testing and QA.

We're not adding real complexity here, we're just raising the bar of our code's quality by showcasing good practices to the community.

Reply:
I agree with using the implementation which showcases a good practice aka encapsulating the logic, and not having to copy paste changes.
Even if it is one line, it makes sense to use the method which helps users to understand a good way of approaching coding in their own projects. It also makes it easier for them in case they want to use scripts here and adjust to their own needs.